src/main/java/Parser.java

```diff
@@ -0,0 +1,130 @@
+import java.io.IOException;
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+public class Parser {
+    protected Storage file;
+    protected TaskList tList;
+
+    /**
+     * Constructor to init a Parser class.
+     * @param file
+     * @param tList
+     */
+    public Parser(Storage file, TaskList tList) {
+        this.file = file;
+        this.tList = tList;
+    }
+
+    /**
+     * Method to process the different inputs that are taken in.
+     * @throws DukeException
+     * @throws IOException
+     */
+    public void parse() throws DukeException, IOException {
+        ArrayList<Task> arr = tList.getArr();
```

Comment:
Maybe this method can be made more modular? It seems that this one method seems to have a lot of functionality that can be broken down into several separate methods, which can be made private for this class if necesary.